go/ai/generate.go

```diff
@@ -1058,3 +1072,177 @@ func handleResumeOption(ctx context.Context, r *registry.Registry, genOpts *Gene
+	}
+
+	// Prepare metadata
+	metadata := map[string]any{
+		"model": map[string]any{
+			"label":    label,
+			"versions": options.Versions,
+			"supports": options.Supports,
+		},
+	}
+
+	if options.ConfigSchema != nil {
+		metadata["model"].(map[string]any)["customOptions"] = options.ConfigSchema
+	}
+	startFunc := func(ctx context.Context, request *ModelRequest) (*core.Operation, error) {
+		startTime := time.Now()
+
+		// Call the user's start function
+		operation, err := start(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add latency metadata
+		if operation.Metadata == nil {
+			operation.Metadata = make(map[string]any)
+		}
+		operation.Metadata["latencyMs"] = float64(time.Since(startTime).Nanoseconds()) / 1e6
+
+		return operation, nil
+	}
+
+	bgAction := core.DefineBackgroundAction[*ModelRequest, *ModelResponse](r, provider, name, *options, metadata, startFunc,
+		func(ctx context.Context, operation *core.Operation) (*core.Operation, error) {
+			return check(ctx, operation)
+		},
+		func(ctx context.Context, operation *core.Operation) (*core.Operation, error) {
+			if cancel == nil {
+				return nil, core.NewError(core.UNIMPLEMENTED, "cancel not implemented")
+			}
+			return cancel(ctx, operation)
+		},
+	)
+
+	return bgAction
+}
+
+// SupportsLongRunning checks if a model supports long-running operations by model name.
+func SupportsLongRunning(r *registry.Registry, modelName string) bool {
+	if modelName == "" {
+		return false
+	}
+
+	provider, name, found := strings.Cut(modelName, "/")
+	if !found {
+		name = provider
+		provider = ""
+	}
+
+	modelInstance := LookupModel(r, provider, name)
+	if modelInstance == nil {
+		return false
+	}
+
+	modelImpl, ok := modelInstance.(*model)
+	if !ok {
+		return false
+	}
+
+	action := (*core.ActionDef[*ModelRequest, *ModelResponse, *ModelResponseChunk])(modelImpl)
+	if action == nil {
+		return false
+	}
+
+	metadata := action.Desc().Metadata
+	if metadata == nil {
+		return false
+	}
+
+	modelMeta, ok := metadata["model"].(map[string]any)
+	if !ok {
+		return false
+	}
+
+	supportsMeta, ok := modelMeta["supports"].(map[string]any)
+	if !ok {
+		return false
+	}
+
+	longRunning, ok := supportsMeta["longRunning"].(bool)
+	if !ok {
+		return false
+	}
+
+	return longRunning
+}
+
+// GenerateOperation generates a model response as a long-running operation based on the provided options.
+// It returns an error if the model does not support long-running operations.
+func GenerateOperation(ctx context.Context, r *registry.Registry, opts ...GenerateOption) (*core.Operation, error) {
+	genOpts := &generateOptions{}
+	for _, opt := range opts {
+		if err := opt.applyGenerate(genOpts); err != nil {
+			return nil, core.NewError(core.INVALID_ARGUMENT, "ai.Generate: error applying options: %v", err)
+		}
+	}
+
+	if !SupportsLongRunning(r, genOpts.ModelName) {
+		return nil, core.NewError(core.UNIMPLEMENTED, "model %q does not support long-running operations", genOpts.ModelName)
```

Comment:
```suggestion
		return nil, core.NewError(core.INVALID_ARGUMENT, "model %q does not support long-running operations", genOpts.ModelName)
```

UNIMPLEMENTED means something else.